src/markdown-pages/collect-data/opentelemetry-manual/custom-span-event.mdx

```diff
@@ -0,0 +1,668 @@
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+provider = TracerProvider(
+    resource=Resource.create({"service.name": "speedeedeebee"})
+)
+provider.add_span_processor(
+    BatchSpanProcessor(
+        OTLPSpanExporter(compression=Compression.Gzip)
+    )
+)
+trace.set_tracer_provider(provider)
+
+tracer = trace.get_tracer(__name__)
+
+class DuplicateKeyError(Exception):
+    pass
+
+class KeyDoesNotExistError(Exception):
+    pass
+
+db = {}
+
+def read(key):
+    """Read key from the database."""
+    global db
+
+    with tracer.start_as_current_span("read", kind=trace.SpanKind.SERVER, record_exception=False, set_status_on_exception=False) as span:
+        try:
+            value = db[key]
+            logging.debug("Successful read")
+            span.set_attribute("key", key)
+            return value
+        except KeyError as ke:
+            msg = f"Key `{key}` doesn't exist"
+            logging.debug(msg)
+            span.add_event("read_key_dne", {"key": key})
+            raise KeyDoesNotExistError(msg)
+
+def create(key, value):
+    """Write key:value to the database."""
+    global db
+
+    with tracer.start_as_current_span("create", kind=trace.SpanKind.SERVER, record_exception=False, set_status_on_exception=False) as span:
+        if key in db:
+            msg = f"Key `{key}` already exists"
+            logging.debug(msg)
+            span.add_event("create_key_exists", {"key": key})
+            raise DuplicateKeyError(msg)
+
+        db[key] = value
+        logging.debug("Successful create")
+        span.set_attribute("key", key)
+        return value
+
+def update(key, value):
+    """Update key in the database."""
+    global db
+
+    with tracer.start_as_current_span("update", kind=trace.SpanKind.SERVER, record_exception=False, set_status_on_exception=False) as span:
+        if key in db:
+            db[key] = value
+            logging.debug("Successful update")
+            span.set_attribute("key", key)
+            return value
+
+        msg = f"Key `{key}` doesn't exist"
+        logging.debug(msg)
+        span.add_event("update_key_dne", {"key": key})
+        raise KeyDoesNotExistError(msg)
+
+def delete(key):
+    """Delete key from the database."""
+    global db
+
+    with tracer.start_as_current_span("delete", kind=trace.SpanKind.SERVER) as span:
+        if key in db:
+            del db[key]
+            logging.debug("Successful delete")
+            span.set_attribute("key", key)
+            return True
+
+        return False
+```
+
+</>
+
+<br />
+
+If you try to update a key that does not exist, your application adds a span event on the span.
+
+</Step>
+
+</Steps>
+
+</Tutorial>
+
+## Restart your simulator
+
+Now that you've changed the application logic, you need to restart your simulator. Make sure you do this in the same terminal window where you set your environment variables.
```

Comment:
It was already mentioned in the last procedure to restart the simulator from the terminal where reader set the environment variable. Should we be mentioning it again?